assets/scss/components/_form.scss

```diff
@@ -21,3 +26,236 @@
         }
     }
 }
+
+.c-text-input,
+.c-password-input,
+.c-textarea-input {
+    display: flex;
+    align-items: center;
+    gap: 0.5rem;
+    padding: 1rem;
+    margin-bottom: 0.5rem;
+
+    border-radius: 0.25rem;
+    border: 0.063rem solid $color-neutrals-75;
+
+    background: $color-white;
+    color: $color-neutrals-500;
+    font-family: $font-sans-serif;
+    font-size: 1rem;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 1.5;
+
+    transition: all 0.2s ease-in-out;
+
+    &:hover {
+        border: 0.063rem solid $color-primary-300;
+        transition: all 0.2s ease-in-out;
+    }
+
+    &:focus {
+        box-shadow: $form-field-box-shadow;
+    }
+}
+
+.c-select-input {
```

Comment:
same here `c-select-input` suggests that there should be a file `components/select-input.scss`

Consider either moving input styling to `components/input.scss` or rename properly :) Also you could move all inputs' common styles to `c-form__input`/`c-input` (depending on which approach you choose) 

Seems like this could be considered a common style - at least at first glance
```
    display: flex;
    align-items: center;
    gap: 0.5rem;
    padding: 1rem;

    border-radius: 0.25rem;
    border: 0.063rem solid $color-neutrals-75;

    background: $color-white;
    color: $color-neutrals-500;
    font-family: $font-sans-serif;
    font-size: 1rem;
    font-style: normal;
    font-weight: 400;
    line-height: 1.5;

     transition: all 0.2s ease-in-out;
```

Also `transition: all 0.2s ease-in-out;` could be replaced with something like `$input-transition` variable. You are using the same transition in 11 places just in this file